napari/layers/utils/plane.py

```diff
@@ -0,0 +1,37 @@
+from typing import Tuple
+
+import numpy as np
+from pydantic import validator
+
+from ...utils.events import EventedModel
+from ...utils.misc import ensure_n_tuple
+
+
+class Plane3D(EventedModel):
+    """Object modelling a plane in 3D with a defined thickness.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A position in 3D, defined in data coordinates.
+    normal_vector : 3-tuple
+        A normal vector in 3D, defined in data coordinates.
+    thickness : float
+        A thickness for the slice
+    """
+
+    position: Tuple[float, float, float] = (0, 0, 0)
+    normal_vector: Tuple[float, float, float] = (1, 0, 0)
+    thickness: float = 10.0
+
+    @validator('position', 'normal_vector', pre=True)
+    def _ensure_3_tuple(cls, v):
+        return ensure_n_tuple(v, n=3)
+
+    @property
+    def normalised_normal_vector(self):
```

Comment:
```suggestion
    def normal_unit_vector(self):
```
I think it is a bit more clear if we call it a unit vector.

Reply:
Or maybe `unit_normal_vector`?

Reply:
I've thought about it a bit more and I think we should always store the normal as a unit vector (i.e., make it a unit vector in the validator). I think many computations one would do with the normal will expect it to be a unit vector.

Reply:
I think you're right, I'll enforce normalisation 👍🏼